### DISCUSSION TITLE
when exporting a --start-at location, it would be a good option to be able to allow other files in other directories to be exported if they are linked from there

### DISCUSSION THREAD

**Discussion by jcolson** [zoni/obsidian-export#76] [Feature requests]
EDITED FOR CLARITY

currently, if --start-at is used, if there are other files in other (../../../SUB)directories, there is no way to export them as well, it would be good to provide that option

**Comment by zoni** [comment]
I'm not sure I understand exactly which behavior you would like to see. Could you please describe this in a little more detail, or provide an example?

**Reply by skewballfox** [reply]
I actually have a mostly implemented solution [here](https://github.com/skewballfox/obsidian-export/tree/recursive-start-at). Right now, all that is left to do is modify the links for the files. It passes the existing test and I'll add some new ones soon. 

Note, I have modified `lib.rs`, in truth I actually started this prior to reading the contributing.md file. I'm not sure implementing this as a postprocessor is feasible due to the need to recursively parse the linked files outside of the `start_at` directory. If you are open to including this in obsidian-export, I can open a PR. Also, if you have an idea of how to implement this as a post-processor, I'm open to reimplementing it.

**Comment by jcolson** [comment]
functionality similar to this:  https://github.com/klalle/ObsidianToHtmlConverter

enabling an option to allow the export of md files/assets linked to from content in the --start-at location that may be in other (../../../SUB)directories that are not included in the --start-at path (../../media/xxx.png, etc) ...

**Comment by skewballfox** [comment]
I found this while trying to figure out how start-at resolved external links.

I'm thinking that the behaviour (toggle-ably) should be:
1. Start at dir supplied
2. Link found outside of that dir
3. Treat that note (or its parent directory) as if it was in the current directory
4. Rercusively apply to a specified depth
5. Once the depth is zero revert to current handling

I think the best way to handle this, without breaking existing behavior is to add a second flag (--recursive, --depth, or whatever) that takes an integer. The default would be 0 so that current start-at behavior won't change.

**Reply by skewballfox** [reply]
If you are open to the idea, I can try and implement this and submit a PR
